TLM/TLM/UI/SubTools/SpeedLimits/Overlay/OverlaySegmentSpeedlimitHandle.cs

```diff
@@ -48,6 +48,24 @@ namespace TrafficManager.UI.SubTools.SpeedLimits.Overlay {
                 netInfo: netInfo,
                 action: action,
                 target: target);
+            Apply(
+                segmentId: this.segmentId_,
+                finalDir: NetInfo.Direction.Both,
+                netInfo: netInfo,
+                action: action,
+                target: target);
+            Apply(
+                segmentId: this.segmentId_,
+                finalDir: NetInfo.Direction.AvoidBackward,
+                netInfo: netInfo,
+                action: action,
+                target: target);
+            Apply(
+                segmentId: this.segmentId_,
+                finalDir: NetInfo.Direction.AvoidForward,
+                netInfo: netInfo,
+                action: action,
+                target: target);
```

Comment:
Each time `Apply()` is called, there's an iteration of all lanes on the segment.

Rather than this series of `Apply()` calls, why not allow `finalDir` to be `null`, and when `null` it applies to all directions except `None`?

Inside `Apply()` method, only time `finalDir` is getting used is here:

```csharp
SpeedLimitManager.Instance.SetSegmentSpeedLimit(segmentId, finalDir, action);
```

Inside that method, the following lines:

```csharp
                NetInfo.Lane laneInfo = segmentInfo.m_lanes[laneIndex];
                NetInfo.Direction d = laneInfo.m_finalDirection;

                if (d == finalDir && laneInfo.MayHaveCustomSpeedLimits()) {
```

Could be tweaked to something like:

```csharp
//...

bool applicableLane =
    (d == finalDir) ||
    (finalDir == null && d != NetInfo.Direction.None);

if (applicableLane && ....
```

This approach would potentially also reduce spamming when setting default speed for a segment (default speed doesn't take a direction, but the multi-`Apply()` approah will be setting default for the segment multiple times).

Note: I've not yet fully invetigated what's happening when multi-segment mode is in play.

Reply:
Also this method in `SpeedLimitManager` would benefit from the `finalDir` being `null` suggestion too: `public bool SetSegmentSpeedLimit(ushort segmentId, SetSpeedLimitAction action) {`

Reply:
> lmk if you want me to send a commit with the suggested changes

If you are sure it won't break anything, go ahead 😉

Reply:
Just noticed few issues, will need to add some tweaks.